internal/executor/resource_executor.go

```diff
@@ -0,0 +1,637 @@
+	result := make(map[string]interface{})
+	// Track original names for collision detection
+	keyOrigins := make(map[string]string)
+
+	for name, resource := range resources {
+		if resource != nil {
+			// Convert snake_case to camelCase
+			key := toCamelCase(name)
+
+			// Check for collision
+			if existingName, exists := keyOrigins[key]; exists {
+				return nil, fmt.Errorf("resource key collision: both %q and %q convert to camelCase key %q", existingName, name, key)
+			}
+
+			keyOrigins[key] = name
+			result[key] = resource.Object
+		}
+	}
+	return result, nil
+}
+
+// isRetryableDiscoveryError determines if a discovery error is transient/retryable
+// (and thus safe to ignore and proceed with create) or fatal (and should fail fast).
+//
+// Retryable errors (log and continue):
+//   - Timeouts (request/server timeouts)
+//   - Server errors (5xx status codes)
+//   - Network/connection errors (connection refused, reset, etc.)
+//   - Service unavailable
+//
+// Non-retryable/fatal errors (fail fast):
+//   - Forbidden (403) - permission denied
+//   - Unauthorized (401) - authentication failure
+//   - Bad request (400) - invalid request
+//   - Invalid/validation errors
+//   - Gone (410) - resource no longer exists
+//   - Other clearly fatal API errors
+func isRetryableDiscoveryError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	// Check for transient Kubernetes API errors (retryable)
+	if apierrors.IsTimeout(err) ||
+		apierrors.IsServerTimeout(err) ||
+		apierrors.IsServiceUnavailable(err) ||
+		apierrors.IsInternalError(err) ||
+		apierrors.IsTooManyRequests(err) {
+		return true
+	}
+
+	// Check for fatal Kubernetes API errors (non-retryable)
+	if apierrors.IsForbidden(err) ||
+		apierrors.IsUnauthorized(err) ||
+		apierrors.IsBadRequest(err) ||
+		apierrors.IsInvalid(err) ||
+		apierrors.IsGone(err) ||
+		apierrors.IsMethodNotSupported(err) ||
+		apierrors.IsNotAcceptable(err) {
+		return false
+	}
+
+	// Check for network-level errors (retryable)
+	if isNetworkError(err) {
+		return true
+	}
+
+	// Default: treat unknown errors as non-retryable to surface issues early
+	return false
+}
+
+// isNetworkError checks if the error is a network-level error (connection issues, DNS, etc.)
+func isNetworkError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	// Check the error chain for network errors
+	var netErr net.Error
+	var opErr *net.OpError
+	var urlErr *url.Error
+
+	// Check for net.Error (includes timeouts)
+	if errors.As(err, &netErr) {
+		return netErr.Timeout()
+	}
+
+	// Check for net.OpError (connection refused, reset, etc.)
+	if errors.As(err, &opErr) {
+		return true
+	}
+
+	// Check for url.Error (URL-related network issues)
+	if errors.As(err, &urlErr) {
+		return urlErr.Timeout() || isNetworkError(urlErr.Err)
+	}
+
+	// Check error message for common network error patterns
+	errMsg := strings.ToLower(err.Error())
+	networkPatterns := []string{
```

Comment:
I would like to avoid this hard coded list if possible. Can you see if we can leverage k8s.io/apimachinery/pkg/util/net or similar to do the same thing. If we use syscall codes we ensure they are universal and stable, string matching can be fragile